python/lsst/pipe/base/pipeline.py

```diff
@@ -997,6 +1030,23 @@ def fromPipeline(
                                          f"({outputNameMapping[name].dimensions}).")
                     composite = DatasetType(name, dsType.dimensions, outputNameMapping[name].storageClass,
                                             universe=registry.dimensions)
+                    # Find all the references in the tasks that match the
+                    # component ref, these will need to have their parent
+                    # storage class set.
+                    toUpdate = []
+                    for other in byTask.values():
+                        # only need to look at inputs, because that is the only
+                        # valid place intermediates can be
+                        if dsType in other.inputs:
+                            toUpdate.append([x for x in other.inputs if x == dsType][0])
+                    # update the parent storage class for thie composit ref
+                    dsType.replaceParentStorageClass(composite.storageClass)
```

Comment:
I thought that was because you said you didn't need a placeholder at all because the rewrite of the graph builder code meant you always knew the parent ahead of time. If you are still using a placeholder I feel we should use the actual placeholder that `DatasetType` knows about so you can have the protection of that code only allowing a true placeholder to be replaced.